org.adempiere.base/src/org/compiere/db/AdempiereDatabase.java

```diff
@@ -333,6 +333,26 @@ public Connection getDriverConnection (String dbUrl, String dbUid, String dbPwd)
 	public boolean forUpdate(PO po, int timeout);
 	
 	public String getNameOfUniqueConstraintError(Exception e);
+	
+	/**
+     * <p>
+     * The "child record found error" contains the 
+     * unique constraint name after the second occurrence
+     * of the opening double quote: ["].
+     * </p>
+     * 
+     * <h3>Example:</h3>
+     * <p>
+     * ERROR: update or delete on table "m_product_category" 
+     * violates foreign key constraint "mprodcategory_mdiscountsbreak" 
+     * on table "m_discountschemabreak" 
+     * Detail: Key (m_product_category_id)=(50000) is still referenced from table "m_discountschemabreak".
+     * </p>
+     * 
+     * @param e
+     * @return constraint name
+     */
+	public String getNameOfChildRecordFoundError(Exception e);

```

Comment:
lets make this obvious and name it as `getForeignKeyConstraint(Exception e)` instead